spec/support/tracer_helpers.rb

```diff
@@ -4,8 +4,39 @@
 
 module TracerHelpers
   # Return a test tracer instance with a faux writer.
-  def get_test_tracer
-    Datadog::Tracer.new(writer: FauxWriter.new)
+  def get_test_tracer(options = {})
+    options = { writer: FauxWriter.new }.merge(options)
+    Datadog::Tracer.new(options).tap do |tracer|
+      # TODO: Let's try to get rid of this override, which has too much
+      #       knowledge about the internal workings of the tracer.
+      #       It is done to prevent the activation of priority sampling
+      #       from wiping out the configured test writer, by replacing it.
```

Comment:
:+1: agree with this comment, this is great, especially if it helps us get by, but a little hacky/nasty.